### ISSUE TITLE
Document configuration options

### ISSUE THREAD

**Issue by barmac** [bpmn-io/dmn-js#688]
__What should we do?__

With dmn-js 12, the library will expose at least two configuration options:

* data types
* expression languages (and defaults)

Let's document this is currently everything is burried in the source code. This could be part of the `usage` section of the readme.

__Why should we do it?__

Let's make the features discoverable.

**Comment by nikku** [comment]
Consider the good practice we talked about in the context of bpmn-js: Have a dedicated `configuration` example that we can link to that showcases how configuration works and configuration basics.

**Comment by nikku** [comment]
Link that example from the readme, where appropriate.

**Comment by CarstenWickner** [comment]
Hi @barmac, @nikku, @pinussilvestrus,

Is there a simple example somewhere for those configuration options?
I've been searching for some time now, but can't find a single one.
I'd be happy with a single JSON example for the "configuration" object.
Mostly the default expression language configuration would be relevant, as we want to reflect the "legacy" engine options (JUEL instead of FEEL by default) also in the newer Modeler component.

Extra question: would these things be configured differently under dmn-js 11 (we haven't braved the breaking changes in version 12 yet) or is the configuration object the same then?

**Comment by barmac** [comment]
Data types: https://github.com/bpmn-io/dmn-js/blob/72145172f6490cd6f30a079d67c75aaa93ff89a4/packages/dmn-js-shared/src/features/data-types/DataTypes.js#L16
Expression languages: https://github.com/bpmn-io/dmn-js/blob/72145172f6490cd6f30a079d67c75aaa93ff89a4/packages/dmn-js-shared/src/features/expression-languages/ExpressionLanguages.js#L21

It hasn't changed since it was implemented.

**Comment by CarstenWickner** [comment]
Okay, cool. Thanks for that.
So it would be something like this, I guess:
```javascript
new DmnJS({
    common: {
        dataTypes: [
            'string',
            'boolean',
            'integer',
            'long',
            'double',
            'date'
        ],
        expressionLanguages: {
            options: [
                { value: 'juel', label: 'JUEL' },
                { value: 'javascript', label: 'JavaScript' },
                { value: 'feel', label: 'FEEL' }
            ],
            defaults: {
                inputCell: 'feel',
                outputCell: 'juel',
                editor: 'juel'
            }
        }
    }
})
```
If I understand it correctly, the `defaults.editor` is a fall-back, i.e., in the above case the matching `default.outputCell` could be omitted.

One extra question though: how to differentiate between an input expression (i.e., in the input column header, that in the past had "juel" as default) vs. an input cell (that in the past had "feel" as default already)?

**Comment by barmac** [comment]
> If I understand it correctly, the defaults.editor is a fall-back, i.e., in the above case the matching `default.outputCell` could be omitted.

Correct.

> One extra question though: how to differentiate between an input expression (i.e., in the input column header, that in the past had "juel" as default) vs. an input cell (that in the past had "feel" as default already)?

If your input expression should have a different default than the editor, you would need to build a custom extension. We don't support that finely grained configuration.

Also, what decision engine do you use?

**Comment by CarstenWickner** [comment]
We're using the Camunda decision engine for years now.
The "legacy" behavior was:
- JUEL as default in input column expressions and output cells
- FEEL as default in input cells

We are just trying to recreate the same behaviour (in order to ensure backward compatibility for the hundreds of stored decision tables) when upgrading to a more recent version of both the engine and the editor component.

**Comment by barmac** [comment]
Indeed, the configuration you provided should maintain the legacy defaults.

If you ever plan to migrate to the new defaults, this will require batch change of the diagrams to set the expression language property for non-FEEL expressions.
